src/test/groovy/graphql/schema/idl/ImmutableTypeDefinitionRegistryTest.groovy

```diff
@@ -169,6 +170,64 @@ class ImmutableTypeDefinitionRegistryTest extends Specification {
         immutableRegistry.remove("key", someDef)
         then:
         thrown(UnsupportedOperationException)
+    }
+
+    def "get implementations of"() {
+        def sdl = serializableSchema + """
+            interface IType {
+                i : String
+            }
+            
+            interface DerivedIType implements IType {
+                i : String
+                d : String                
+            } 
+            
+        """
+        for (int i = 0; i < 10; i++) {
+            sdl += """
+                type OT$i implements IType {
+                    i : String
+                }
+                """
+
+        }
+        for (int i = 0; i < 5; i++) {
+            sdl += """
+            type DT$i implements DerivedIType {
+                i : String
+                d : String                
+            } 
+                """
+
+        }
+        def immutableRegistry = new SchemaParser().parse(sdl).readOnly()
+
+        Map<String, InterfaceTypeDefinition> interfaces = immutableRegistry.getTypesMap(InterfaceTypeDefinition.class)
+
+        when:
+        def iFieldType = interfaces.get("IType")
+        def allImplementationsOf = immutableRegistry.getAllImplementationsOf(iFieldType)
+        def implementationsOf = immutableRegistry.getImplementationsOf(iFieldType)
+
+        then:
+        allImplementationsOf.size() == 11
+        allImplementationsOf.collect({ it.getName() }).every { it.startsWith("OT") || it == "DerivedIType" }
+
+        implementationsOf.size() == 10
+        implementationsOf.collect({ it.getName() }).every { it.startsWith("OT") }
+
+        when:
+        def iDerivedType = interfaces.get("DerivedIType")
+        allImplementationsOf = immutableRegistry.getAllImplementationsOf(iDerivedType)
+        implementationsOf = immutableRegistry.getImplementationsOf(iDerivedType)
+
+        then:
+        allImplementationsOf.size() == 5
+        allImplementationsOf.collect({ it.getName() }).every { it.startsWith("DT") }
+
+        implementationsOf.size() == 5
+        implementationsOf.collect({ it.getName() }).every { it.startsWith("DT") }
```

Comment:
This is just to make sure we did what we thought we would do and I can debug and confirm the pre-built map is as expected and its going to be cheaper when a full schema build is done and repeated calls are made on it to check interfaces and types

Reply:
However I killed the cache code but I figured this code is still valid